R/bootstrap.R

```diff
@@ -687,6 +774,43 @@ tabPanelBody <- function(value, ..., icon = NULL) {
   tabPanel(title = NULL, ..., value = value, icon = icon)
 }
 
+isTabPanel <- function(x) {
+  tagHasClass(x, "tab-pane")
+}
+
+#' @rdname tabPanel
+#' @export
+navLink <- function(title, href, ..., icon = NULL, target = "_blank") {
+  args <- list2(
+    href = href, target = target,
+    icon, title, ...
+  )
+  link <- tags$a(!!!dropNulls(args))
+  class(link) <- c("shiny_nav_link", class(link))
+  link
+}
+
+isNavLink <- function(x) {
+  inherits(x, "shiny_nav_link")
+}
+
+#' @rdname tabPanel
+#' @export
+navForm <- function(...) {
+  form <- tags$form(class = "form-inline", ...)
+  class(form) <- c("shiny_nav_form", class(form))
+  form
+}
```

Comment:
Could we combine these into a single function (navItem())?

Reply:
If we have `navItem()`, then links could be done with:

```R
navItem(tags$a(href = "...", icon(...), "Label"))
```

Also, to be correct HTML, each `navItem` would have to be an `<li>`.